ext/spl/spl_fixedarray.c

```diff
@@ -102,13 +102,18 @@ static void spl_fixedarray_init_elems(spl_fixedarray *array, zend_long from, zen
 	}
 }
 
+static void spl_fixedarray_init_non_empty_struct(spl_fixedarray *array, zend_long size)
+{
+	array->size = 0; /* reset size in case ecalloc() fails */
+	array->elements = safe_emalloc(size, sizeof(zval), 0);
```

Comment:
```suggestion
	array->elements = size ? safe_emalloc(size, sizeof(zval), 0) : NULL;
```

I just realized - this will call emalloc with size 0 which will return a non-empty pointer

This is something spl_fixedarray_init avoided
```
(gdb) print _emalloc(0)
$1 = (void *) 0x7ffff3802000
```